### PR TITLE
Feature/field description

### PR DESCRIPTION
**Pull request by DanRibbens**
## Description

- Adds description to all field types
- Adds description to Globals and Collections

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/admin/components/forms/RenderFieldDescription/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import './index.scss';
+
+const RenderFieldDescription: React.FC<Props> = (props) => {
+  const {
+    description,
+    value,
+    CustomComponent,
+  } = props;
+
+  if (CustomComponent) {
```
Comment (DanRibbens) [inline src/admin/components/forms/RenderFieldDescription/index.tsx:12]: Here I am prioritizing a CustomComponent over `admin.description` from the config.

File: src/fields/config/types.ts [hunk 2]
```diff
@@ -31,7 +31,13 @@ type Admin = {
   readOnly?: boolean;
   disabled?: boolean;
   condition?: Condition;
-  components?: { [key: string]: React.ComponentType };
+  description?: string | ((data: Record<string, unknown>) => string);
+  components?: {
+    Filter?: React.ComponentType;
+    Cell?: React.ComponentType;
+    Field?: React.ComponentType;
+    Description?: React.ComponentType<{value: unknown}>
```
[the remark this answers is not in the document]
Comment (jmikrut) [reply src/fields/config/types.ts:39]: @DanRibbens can we leave it _**only**_ as a Component? It certainly feels redundant at first glance to have it avaliable from so many places. I think it's perfectly acceptable to prefer this pattern and call it a day (removing the `admin.description`.

**Comment by Ontopic** [comment]
Any chance of this being released today (Tuesday morning in Europe now)? Big client coming in tomorrow for a demo of the CMS, would be nice to quickly put this in so they understand better what they're clicking around in :) Or safe enough at least to fork this pr?

**Comment by DanRibbens** [comment]
This will be released today, likely about 6 hours from now (noon our time). It is safe to use this PR, let us know if you do and find any issues.

File: docs/fields/overview.mdx [hunk 3]
```diff
@@ -138,3 +139,27 @@ The `condition` function should return a boolean that will control if the field
 ### Custom components
 
 All Payload fields support the ability to swap in your own React components with ease. For more information, including examples, [click here](/docs/admin/components#fields).
+
+### Description
+
+You can simply provide a string that will show by the field, but there are use cases where you may want to create some dynamic feedback. By using a function for the `description` property you can provide rich feedback in realtime the user interacts with the form.
+
+The `description` can be assigned a function can wth the parameter `{ value }` and can return the `string` you wish to display.
```
Comment (jmikrut) [inline docs/fields/overview.mdx:147]: Does this _**need**_ to return a string? Shouldn't it still work if you returned a React component?

File: src/admin/components/forms/field-types/DateTime/index.tsx [hunk 4]
```diff
@@ -80,6 +82,10 @@ const DateTime: React.FC<Props> = (props) => {
           value={value as Date}
         />
       </div>
+      <FieldDescription
+        value={value || new Date()}
```
Comment (jmikrut) [inline src/admin/components/forms/field-types/DateTime/index.tsx:86]: Do we need to pass in this `new Date()`?

File: src/admin/components/forms/field-types/Email/index.tsx [hunk 5]
```diff
@@ -80,6 +82,10 @@ const Email: React.FC<Props> = (props) => {
         name={path}
         autoComplete={autoComplete}
       />
+      <FieldDescription
+        value={value || ''}
```
Comment (jmikrut) [inline src/admin/components/forms/field-types/Email/index.tsx:86]: Seems redundant to have to pass the value as empty string - here, and everywhere. Could be better pattern to handle within the component itself.

File: src/collections/config/schema.ts [hunk 6]
```diff
@@ -34,8 +34,7 @@ const collectionSchema = joi.object().keys({
     preview: joi.func(),
     disableDuplicate: joi.bool(),
   }),
-  fields: joi.array()
-    .items(fieldSchema),
+  fields: joi.array(),
```
Comment (jmikrut) [inline src/collections/config/schema.ts:37]: What is this change for?

File: src/collections/config/schema.ts [hunk 7]
```diff
@@ -24,6 +23,7 @@ const collectionSchema = joi.object().keys({
   admin: joi.object({
     useAsTitle: joi.string(),
     defaultColumns: joi.array().items(joi.string()),
+    description: joi.string(),
```
Comment (jmikrut) [inline src/collections/config/schema.ts:26]: How does this accept a function that returns a string?

File: src/collections/config/types.ts [hunk 8]
```diff
@@ -99,7 +99,8 @@ export type PayloadCollectionConfig = {
   admin?: {
     useAsTitle?: string;
     defaultColumns?: string[];
-    disableDuplicate?: boolean
+    description?: string;
```
Comment (jmikrut) [inline src/collections/config/types.ts:102]: Shouldn't this be `string | (val: unknown) => string | React.Component` ?

File: src/globals/config/types.ts [hunk 9]
```diff
@@ -54,6 +54,7 @@ export type PayloadGlobalConfig = {
   }
   fields: Field[];
   admin?: {
+    description?: string,
```
Comment (jmikrut) [inline src/globals/config/types.ts:57]: This should also be `string | (val: unknown) => string | React.ComponentType` AFAIK

**Comment by Ontopic** [comment]
Right on time for me to work in the most important ones. Thanks 🤗. This client may take several licenses in the future, so worth giving it the best presentation ;)

**Comment by DanRibbens** [comment]
That is music to our ears @Ontopic!

**Pull request merged**